lib/bolt_server/base_config.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require 'hocon'
+require 'bolt/error'
+
+module BoltServer
+  class BaseConfig
+    def self.base_config_keys
```

Comment:
I think there's a better way to do this, but I wasn't able to turn up how.
What I want is to set a 'default' here that's available in for the child subclass to modify, and when the function is called the child class modified version is called.

Reply:
You can use `super` from the child class to call the parent implementation.

```rb
# parent
def config_keys
  [ ... ]
end

# child
def config_keys
  super + ['additional_key']
end
```

Reply:
I guess it's a little more complicated because that's defined as a class method. But I think just making it an instance method would be fine.

Reply:
I wasn't sure how to create the accessor methods with config keys as an instance method:
```
    config_keys.each do |key|
      define_method(key.tr('-', '_').to_sym) do
        @data[key]
      end
    end
```

With that I get `undefined method 'config_keys' for PlanExecutor::Config:Class`